### PR TITLE
browser(firefox): roll Firefox Stable to 102

### PR DESCRIPTION
**Pull request by aslushnikov**
This roll:
- NetworkObserver now uses the `remote's` ChannelEventSink layer to
  subscribe to redirects.
- Wheel events now must be dispatched from browser process.
- There's a new API for console messages
- The old methods to wait for search service and addon manager no longer
  work; speculatively remove them since neither `remote` nor
  `marionette` have anything like this.

Native manual merge: https://github.com/aslushnikov/juggler/commit/9e6fcfd868a447539095398ae68469f37339cbc1

### CODE REVIEW

File: tests/library/browsercontext-cookies.spec.ts [hunk 1]
```diff
@@ -116,9 +116,10 @@ it('should get multiple cookies', async ({ context, page, server, defaultSameSit
     document.cookie = 'password=1234';
     return document.cookie.split('; ').sort().join('; ');
   });
-  const cookies = new Set(await context.cookies());
+  const cookies = await context.cookies();
   expect(documentCookie).toBe('password=1234; username=John Doe');
-  expect(cookies).toEqual(new Set([
+  cookies.sort((c1, c2) => c1.name < c2.name ? -1 : 1);
```
Comment (dgozman) [inline tests/library/browsercontext-cookies.spec.ts:121]: I wonder why this change? Seems equivalent :)

File: browser_patches/firefox/juggler/content/Runtime.js [hunk 2]
```diff
@@ -182,6 +183,30 @@ class Runtime {
     this._eventListeners.push(() => Services.obs.removeObserver(consoleObserver, "console-api-log-event"));
   }
 
+  _registerConsoleAPIListener(Services) {
```
Comment (dgozman) [inline browser_patches/firefox/juggler/content/Runtime.js:186]: Does this mean we should remove some other console handler?

**Review by dgozman — APPROVED** [review]
- Why didn't we do these changes when rolling beta to 102?
- Removing addon manager waiting looks scary :)

File: tests/library/browsercontext-cookies.spec.ts (hunk 1 above)
Comment (aslushnikov) [reply tests/library/browsercontext-cookies.spec.ts]: indeed! I was under impression that Set's aren't supported

**Comment by aslushnikov** [comment]
> Why didn't we do these changes when rolling beta to 102?

@dgozman I was under impression that i was hitting cross-process navigation, thus started working on Fission

> Removing addon manager waiting looks scary :)

It surely does!

File: tests/library/browsercontext-cookies.spec.ts (hunk 1 above)
Comment (aslushnikov) [reply tests/library/browsercontext-cookies.spec.ts]: reverted.
